modvendor/modvendor_test.go

```diff
@@ -49,19 +874,23 @@ func TestModVendorWithCommitIDRef(t *testing.T) {
 	gitURL := "file://" + repoSandbox.RootDir()
 	rootdir := t.TempDir()
 
-	const vendordir = "/dir/reftest/vendor"
-	cloneDir, err := modvendor.Vendor(rootdir, vendordir, tf.Source{
-		URL:  gitURL,
-		Ref:  ref,
-		Path: path,
-	})
+	source, err := tf.ParseSource(fmt.Sprintf("git::%s?ref=%s", gitURL, ref))
```

Comment:
hmm any reason why we need to encode in a source and then parse it right away instead of creating the tf.Source directly ?

Reply:
less code? also if we change the way we encode the source later (or add new fields), we have less places to fix, and the tf.ParseSource() will fill all fields and catch errors, etc.

Reply:
hmm create one struct passing 3 fields VS calling Sprintf + a parseing function...Im not so sure it is less code :thinking:. But not a strong opinion on this...on the spirit of even less code would extract this into a function that already checks the error + does the printf.

I get the point of preparing for future changes...it just feels funny that you already have parsed information, then you glue it together to parse it again..and Im not so sure if this will really change or if we will have wild information being extracted from the parsing in the future.

Reply:
Just to explain better at least the point already more code. Initializing the source struct directly in a smilar way to how the parsing option is done would be done like this:

```go
source := tf.Source{ URL:  gitURL, Ref:  ref, Path: path }
funcall(source)
```

It is a single line to create the source with information already available + call something that uses it. The current code was passing it directly on the function call and then we break into multiple lines, but if we compare with the parsing approach that has its own line it would be a more fair comparison. It is also one line and it involves only a simple struct intialization, IMHO less code than this:

```go
source, err := tf.ParseSource(fmt.Sprintf("git::%s?ref=%s", gitURL, ref))
assert.NoError(t, err)
funcall(source)
```

On top of the extra error handling, even though it is a single line you have both a printf + a parsing happening on this single line, IMHO that is more code.

But depending on future changes it could happen that the parsing function could already fill out fields for us, but that is more related to the idea of having a test specific constructor that abstract details than having to parse things necessarily, but could be an advantage. So in general I'm not pushing for not using the parse function, but I wanted to defend a little why I said it is more code =P